esmvaltool/diag_scripts/autoassess/autoassess_source/rms.py

```diff
@@ -0,0 +1,314 @@
+"""
+Port for ESMValTool v2 from v1
+
+Uses: ESMValTool v2, Python 3.x
+Valeriu Predoi, UREAD, July 2018
+
+Functionality: computes root mean squares for a bunch of geographical
+regions;
+
+Original docstring:
+This file contains an rms class which contains all the information needed
+to make a rms table of a particular region.
+"""
+
+import os
+import math
+import logging
+import numpy.ma as ma
+import iris
+from .valmod_radiation import area_avg
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+class RMSLISTCLASS(list):
+    """
+    Construct the regions class
+
+    This is the class for a list of RMSCLASS (i.e. for lots of regions)
+    """
+
+    def __init__(self, *args):
+        """Init"""
+        if not args:
+            super(RMSLISTCLASS, self).__init__()
+        else:
+            super(RMSLISTCLASS, self).__init__(args[0])
+
+    def __repr__(self):
+        """Repr"""
+        rms_out = "["
+        for rms_item in self:
+            rms_out += "rms.RMSCLASS for " + rms_item.region + ", \n"
+        rms_out = rms_out[0:-3]
+        rms_out += "]"
+
+        return rms_out
+
+    def __call__(self, region=False):
+        """Call"""
+        rms_found = False
+        region_list = []
+        for rms_item in self:
+            region_list.append(rms_item.region)
+            if region:
+                if rms_item.region == region:
+                    rms_returned = rms_item
+                    rms_found = True
+        if not region:
+            logger.warning(
+                "Please supply a region using the region='xxx' input. " +
+                "Available regions are:")
+        elif not rms_found:
+            logger.warning("ERROR: Requested region not found.")
+        if not rms_found:
+            logger.error(region_list)
+            raise Exception
+        return rms_returned
+
+
+# This is the class for one set of rms values (i.e. for one region)
+class RMSCLASS:
+    """Class per region"""
+
+    def __init__(self, region, exper='experiment', control='control'):
+        """
+        Create instances of this class but also start making
+
+        html files that will contain all the rms data. (old)
+
+        region = the region name
+        exper = experiment jobid
+        control = control jobid
+        """
+        # Store the region name, experiment and control
+        self.region = region
+        self.exper = exper
+        self.control = control
+
+        # TODO: Store following information in dictionary
```

Comment:
If you could do this TODO I'm happy as far as Codacy is concerned.

Reply:
blehh, that's not my TODO, it originates from whoever wrote the initial code, man, I can just delete it since it works fine like this :grin:

Reply:
It works fine but it hurts my eyes ;-)